TanoaPeakSimulator/TanoaPeakSimulator/Tanoa.py

```diff
@@ -0,0 +1,128 @@
+import secrets
+import random
+
+
+def main():
+    location_length = 1000
+    bp_length = 100
+    bp_amount = 200
+    # This should be in a function with the file as a parameter for when this gets turned into
+    genome_file = open("dmel-all-chromosome-r6.38.fasta", "r")  # code which takes arguments
+    sequences = genome_file.read()
+    sequences_arr = sequences.split('>')
+    sequence_object_array = []
+    for sequence in sequences_arr:
+        sequence_split = sequence.splitlines()
+        if len(sequence_split) != 0:
+            header = sequence_split[0]
+            body = ""
+            for x in range(1, len(sequence_split)):
+                body += sequence_split[x]
+            sequence_object_array.append(Chromosome(header, body))
+
+    # Code for running all of the chromosomes. Commented out for now since I don't have enough memory...
+    # for chromosome in sequence_object_array:
+    #     for x in range(2000):
+    #         get_random_base_pair_string(location_length, bp_length, chromosome.sequence,
+    #                                     chromosome.bp_segment_arr)
+    # for chromosome in sequence_object_array:
+    #     chromosome.small_bp_segment_list = get_random_base_pair_string_list(chromosome.bp_segment_arr, bp_length,
+    #                                                                         bp_amount, chromosome.sequence)
+    chromosome = sequence_object_array[1]
+    for x in range(100):
+        get_random_base_pair_string(location_length, bp_length, chromosome.sequence, chromosome.bp_segment_arr)
+
+    chromosome.small_bp_segment_list = get_random_base_pair_string_list(chromosome.bp_segment_arr, bp_length,
+                                                                        bp_amount, chromosome.sequence)
+
+    generate_output_files(chromosome)
+
+
+# Gets a psudo-random string of a given amount of characters from the main sequence. This will move to taking
+# the buffers on the sides and saving that as well so when sampled later we don't have to return to the
+# Very large string. That's inefficient.
+# Parmeters: sequence_length: Length of the big sample that you want to take
+# bp_length: the length of a smaller base pair that you will sample from the broader region
+#   this is setup so that the buffer can be dynamic based on the size of the reads you want to match
+#   to the original chromosome are
+# Sequence: The actual character sequence, generally attached to a chromosome like 2L or 2R
+# array_of_segments: array of BasePairSegments (samples) from the original chromosome.
+#   This array is stored within a chromosome as bp_sequence_array and is modifying that object.
+def get_random_base_pair_string(sequence_length, bp_length, sequence, array_of_segments):
+    if len(sequence) > sequence_length + 101:  # Ignore sequences that don't have enough length to work...
```

Comment:
Do the 101 and 99 need to be hard-coded or can we pass in the region widths as parameters?

Reply:
That line there for now is so that the sequences in the file that aren't long enough to even sample (in my case at 1000 bps long) don't get sampled and have errors getting thrown. it should be bp_length + 1 for the if statement and the 99 should be bp_length - 1. I had the second one changed in my local file and haven't pushed that out yet. I still need to figure out how I'm going to do that though since the issue I'm coming up on is that if I want 1000 length samples, I end up actually needing 1000 + 99 + 99 length samples since the front and back have those buffer zones where we allow the potential reads to overlap by at least 1 bp.